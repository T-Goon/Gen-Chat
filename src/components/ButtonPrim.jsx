import React from "react";
import { TouchableOpacity, Text, StyleSheet } from "react-native";

const ButtonPrim = (props) => {
    const { onPress, buttonStyles } = props;

    return (
        <TouchableOpacity style={{...styles.button, ...buttonStyles}} onPress={onPress}>
            <Text style={styles.text}>{props.children}</Text>
        </TouchableOpacity>
    );
};

const styles = StyleSheet.create({
    button: {
      backgroundColor: 'hsl(550, 100%, 50%)',
      alignItems: 'center',
      justifyContent: 'center',
      paddingVertical: 8,
      paddingHorizontal: 16,
      borderRadius: 10,
      alignSelf: 'baseline',
    },
    text: {
      fontSize: 20,
      fontWeight: 'bold'
    }
  });

export default ButtonPrim;