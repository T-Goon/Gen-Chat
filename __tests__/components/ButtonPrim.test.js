import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

import ButtonPrim from '../../src/components/ButtonPrim';

describe('<ButtonPrim/ >', () => {
    it('Num Children', () => {
        const buttonPrim = render(<ButtonPrim />);
        const json = buttonPrim.toJSON();

        expect(json.children.length).toBe(1);
    });

    it('OnPress Prop', () => {
        const onPress = jest.fn();
        const buttonPrim = render(<ButtonPrim onPress={onPress} />);

        fireEvent(buttonPrim.container, 'press');
        expect(onPress.mock.calls.length).toBe(1);
    });

    it('Snapshot', () => {
        const buttonPrimJson = render(<ButtonPrim />).toJSON();
        
        expect(buttonPrimJson).toMatchSnapshot();
    });
})