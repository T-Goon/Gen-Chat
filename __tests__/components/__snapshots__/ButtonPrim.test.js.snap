// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonPrim/ > Snapshot 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "baseline",
      "backgroundColor": "hsl(550, 100%, 50%)",
      "borderRadius": 10,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  />
</View>
`;
